src/Framework/DomQuery.php

```diff
@@ -81,7 +81,16 @@ public static function css2xpath($css)
 		$xpath = '//*';
 		preg_match_all('/
 			([#.:]?)([a-z][a-z0-9_-]*)|               # id, class, pseudoclass (1,2)
-			\[([a-z0-9_-]+)(?:([~*^$]?)=([^\]]+))?\]| # [attr=val] (3,4,5)
+			\[
+				([a-z0-9_-]+)
+				(?:
+					([~*^$]?)=(
+						(?:"[^"]*")|
+						(?:\'[^\']*\')|
+						(?:[^\]]+)
+					)
+				)?
+			\]|                                       # [attr=val] (3,4,5)
 			\s*([>,+~])\s*|                           # > , + ~ (6)
 			(\s+)|                                    # whitespace (7)
 			(\*)                                      # * (8)
```

Comment:
@dg I noticed that `$m[8]` is not used anywhere. Not sure how are the tests passing :)